;;; TOOL: run-interp-spec
;;; STDIN_FILE: third_party/testsuite/proposals/exception-handling/try_table.wast
;;; ARGS*: --enable-exceptions --enable-tail-call
(;; STDOUT ;;;
out/test/spec/exception-handling/try_table.wast:261: assert_trap passed: unreachable executed
out/test/spec/exception-handling/try_table.wast:264: assert_trap passed: integer divide by zero
out/test/spec/exception-handling/try_table.wast:268: assert_exception passed
out/test/spec/exception-handling/try_table.wast:272: assert_exception passed
out/test/spec/exception-handling/try_table.wast:310: assert_exception passed
out/test/spec/exception-handling/try_table.wast:311: assert_exception passed
out/test/spec/exception-handling/try_table.wast:340: assert_malformed passed:
  out/test/spec/exception-handling/try_table/try_table.3.wat:1:16: error: unexpected token "catch_all", expected an instr.
  (module (func (catch_all)))
                 ^^^^^^^^^
out/test/spec/exception-handling/try_table.wast:345: assert_malformed passed:
  out/test/spec/exception-handling/try_table/try_table.4.wat:1:25: error: unexpected token "catch", expected an instr.
  (module (tag $e) (func (catch $e)))
                          ^^^^^
out/test/spec/exception-handling/try_table.wast:360: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.6.wasm:000001c: error: type mismatch in try table, expected [i32] but got []
  000001c: error: OnEndExpr callback failed
out/test/spec/exception-handling/try_table.wast:364: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.7.wasm:000001e: error: type mismatch in try table, expected [i32] but got [i64]
  000001e: error: OnEndExpr callback failed
out/test/spec/exception-handling/try_table.wast:369: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.8.wasm:0000022: error: catch signature doesn't match target: expected [exnref], got []
  0000022: error: OnTryTableExpr callback failed
out/test/spec/exception-handling/try_table.wast:373: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.9.wasm:0000026: error: catch signature doesn't match target: expected [], got [exnref]
  0000026: error: OnTryTableExpr callback failed
out/test/spec/exception-handling/try_table.wast:377: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.10.wasm:000001c: error: catch signature doesn't match target: expected [exnref], got []
  000001c: error: OnTryTableExpr callback failed
out/test/spec/exception-handling/try_table.wast:381: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.11.wasm:000001d: error: catch signature doesn't match target: expected [], got [exnref]
  000001d: error: OnTryTableExpr callback failed
out/test/spec/exception-handling/try_table.wast:385: assert_invalid passed:
  out/test/spec/exception-handling/try_table/try_table.12.wasm:0000028: error: catch signature doesn't match target: expected [i64, exnref], got [i32, exnref]
  0000028: error: OnTryTableExpr callback failed
53/53 tests passed.
;;; STDOUT ;;)
